### PR TITLE
Change map-marker with styleable svg

### LINKED ISSUE #4: Can I change color a marker
Can I change color a marker

### PR DESCRIPTION
**Pull request by xstable**
Close #4

**Please remember** to flag my PR with Label `hacktoberfest-accepted` as litte thank you. ;)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -34,6 +34,11 @@ export default class Marker extends Component {
     latLngToPixel: PropTypes.func,
     pixelToLatLng: PropTypes.func
   }
+  static defaultProps = {
+    markerSize: '30',
```
Comment (mariusandra) [inline src/index.js]: The markerSize default prop is a `"string"`, but the PropTypes say it should be a `PropTypes.number`

File: src/index.js [hunk 2]
```diff
@@ -102,22 +88,35 @@ export default class Marker extends Component {
   // render
 
   render () {
-    const { left, top, onClick } = this.props
+    const { left, top, onClick, markerColor, markerColorHover, markerSize } = this.props
+    const { hover } = this.state
+    const svgSize = {width: markerSize, height: markerSize}
```
Comment (mariusandra) [inline src/index.js]: I don't understand this `svgSize`. Where is it used?

**Review by mariusandra — CHANGES_REQUESTED** [review]
Hey, some code comments inline and then two bigger issues:
- `yarn build` fails now without the `src/img/` folder
- it would be great if the marker would look the same as it did before.

Before:
image

After:
image

**Comment by xstable** [comment]
> * it would be great if the marker would look the same as it did before.

@mariusandra 
I'm not such a good designer with svg. So it would be fine, if you take this PR for the general technical implementation of a svg where the color is changeable by parameters... and create another issue (for a more skilled Designer) to enhance the SVG themself.

Anyway, I've tried to optimize the svg a bit by adding a shadow which color can be adjust now via parameter.

**Tipp:**  to be able to change the color, it's important that the SVG has "path"-Tags inside, (and not embedded PNG).

File: src/index.js (hunk 1 above)
Comment (xstable) [reply src/index.js]: Fixed that

File: src/index.js (hunk 2 above)
Comment (xstable) [reply src/index.js]: I've fixed that.

**Comment by xstable** [comment]
@mariusandra should be fine now. Please have a look at it.

**Comment by xstable** [comment]
@mariusandra all fine now, or any suggestions?
Would be nice if you could merge it soon, because I also need this feature for a current project.
Thnx!

**Comment by mariusandra** [comment]
Hey! Apologies for the delay, been super busy :). I'll review it soon, tonight hopefully.

**Comment by mariusandra** [comment]
Hey, unfortunately this needs more work.

First, the markers don't anchor to a coordinate:


And I'd still like them to have the more unique style they had before. 
image

I'm not good in SVG either, but it's a matter of loading the png into figma, tracing it with path tools three times (once for the white outline, once for the greeish inner, once for the ball), making sure the right colors and styles are applied and then exporting that as a `<svg>` to copy/paste to react.

Unfortunately I can't merge this before those two points are fixed because otherwise I'll need to do extra work before I can cut a new version of this package again. Work that I don't have time for in October, sorry :/.

If you do want to get a PR out quickly to count towards your hacktoberfest quota, feel free to make something simple. For example upgrade the demo in this repository to the last version of pigeon-maps and convert `demo/demo.js` a functional (hooks-based) component.

**Comment by xstable** [comment]
Thanks for the response. I don't have an figma-account. So feel free to do it in November if you like.
In the meantime I have to use my patched version in my project. Will have a look at the placement-issue later on, if I have some more spare time. Anybody who wan't to give helping hands for the current issue is very welcome.

**Comment by xstable** [comment]
@mariusandra I see now the issue with the marker-position. It only appears if you change the Marker-Size by props.
So it seems I have to calculate the `imageOffset` by depending on the given props-width (see Line 4 https://github.com/xstable/pigeon-marker/blob/master/src/index.js#L4).

For the SVG, I've created this new one, which should get closer to your needs: 

```svg
<svg viewBox="0 0 24 24" style="width: 30px; height: 30px;" xmlns="http://www.w3.org/2000/svg" xmlns:bx="https://boxy-svg.com">
  <defs></defs>
  <filter id="shadow">
    <feDropShadow dx="0" dy="0" stdDeviation="0.9" flood-color="#fafafa"></feDropShadow>
  </filter>
  <ellipse style="stroke-opacity: 0.65; paint-order: fill; fill: rgb(139, 139, 139); fill-opacity: 0.8;" cx="11.961" cy="22.541" rx="6.058" ry="1.431" bx:origin="0.77789 -3.095136">
    <title>Shaddow</title>
  </ellipse>
  <path d="M 12 1.017 C 7.905 1.017 4.584 4.218 4.584 8.169 C 4.584 13.53 12 21.449 12 21.449 C 12 21.449 19.415 13.53 19.415 8.169 C 19.415 4.218 16.095 1.017 12 1.017 Z" style="filter: url(#shadow); stroke-miterlimit: 1; paint-order: stroke; stroke-width: 2px; stroke-linecap: square; stroke-opacity: 0.9; stroke: rgb(255, 255, 255);" fill="#f00" bx:origin="0.631861 0.519651"></path>
  <circle style="fill: rgb(255, 255, 255); stroke-opacity: 0.85; fill-opacity: 0.85; paint-order: fill;" cx="11.46" cy="10.761" r="2.122" transform="matrix(1.267309, 0, 0, 1.22227, -2.505215, -4.98817)" bx:origin="1.115508 1.285579">
    <title>InnerCircle</title>
  </circle>
</svg>
```

**Comment by xstable** [comment]
@mariusandra  
Changed the SVG image and fixed the issue with anchor to coordinate. Plz give it a try.

Close #4

**Comment by xstable** [comment]
@mariusandra any progress here? Would you like to merge it?

**Comment by mariusandra** [comment]
Hey @xstable ! 
Thank you for all the work you have put in to this PR! I really appreciate it!

Unfortunately there are still some outstanding issues:

1) On retina screens the marker is 2x the width/height causing issues like on the screencast/gif above. 
2) The terms `markerColorHover` vs `markerShadowColor` are not very consistent.
3) The marker icon is still different from what came before.

All of these can be overcome, yet I've decided to go a different route. 

I have [recreated the marker](https://github.com/mariusandra/pigeon-maps/blob/master/src/marker.tsx) inside the pigeon-maps library itself, using a SVG that looks like the old one and is rewritten in the newer react functional style.

I plan to add a few extra widgets (infowindow, overlay and draggable overlay) before releasing it as a new version of pigeon-maps. This way all you need to do is to run `import { Map, Marker } from 'pigeon-maps'` and do what is needed. In the future I plan to add a few other other components (e.g. polylines) and extend the "standard library" of pigeon-maps components even further.

It will still be a few more days before I'm ready to release all of that as a new version though. In the mean time, feel free to fork and release the original `pigeon-marker` as a new project to cover your own needs.

**Pull request closed without merging**